howso.amlg

```diff
@@ -790,25 +790,32 @@
 	(retrieve_from_entity label)
 
 	;create a return response object in the format of:
-	; {
-	;	'status' : 'ok', # string 'ok' or 'error'
-	;	'errors' : [ {'detail': 'Some error message'} ], # list of error dicts
-	;	'warnings' : [ {'detail': 'Some warning message'} ], # list of warning dicts
-	;	'payload' : whatever # json response payload
-	; }
+	; success:
+	; [ 1, { 'payload' : payload, 'warnings' : warnings } ]
+	;
+	; error:
+	; [ 0, { 'details': error_descriptions, 'code': error_code, 'errors': field_validation_details } ]
 	;
 	;parameters:
 	; errors: optional, list of error strings to output
+	; error_code: optional, string defining the error code, e.g., 'validation'
+	; error_details: optional, assoc defining field validation errors
 	; warnings: optional, list of warning strings to output
 	; payload: any value or object
 	#!Return
 	(if errors
-		(assoc "status" "error"	"errors" (map (lambda (assoc "detail" (current_value 1))) errors) "warnings" (null)	"payload" payload)
+		(list 0
+			(append
+				(assoc "details" (map (lambda (current_value )) errors) )
```

Comment:
The error response should include a key `detail` that is a string or list of strings. Optimally, it should only be a list if there are more than one message.